lxd/db/images.go

```diff
@@ -464,11 +464,15 @@ func (c *Cluster) GetImage(project, fingerprint string, public bool) (int, *api.
 		if !enabled {
 			project = "default"
 		}
-		images, err := tx.GetImages(ImageFilter{
-			Project:     project,
-			Fingerprint: fingerprint + "%",
-			Public:      public,
-		})
+
+		// FIXME
+		// Properly check zero values of 'public', and don't include it in GetImage at all if it should be ignored.
```

Comment:
@masnax IIRC this is where the bug crept in before, is it worth adding a unit test so we can test this function `GetImage` always does the 'correct' thing, even after the various refactors we are doing to it?

Reply:
@masnax is it a big change to rename the `public` argument to `publicOnly`, as I think this would make it a lot clearer what that argument does? Along with a comment update to explain what that argument does.

Reply:
A couple more things while we are looking at this function:

1. We don't use the function argument type compact format, i.e `project, fingerprint string` instead please could you update to `project string, fingerprint string`.
2. I suppose we should change the `fingerprint` argument to `fingerprintPrefix` for clarity, as that is what it is being used as in this function.

Reply:
> @masnax is it a big change to rename the `public` argument to `publicOnly`, as I think this would make it a lot clearer what that argument does? Along with a comment update to explain what that argument does.

What are your thoughts on a mask function that doesn't take the `public` argument at all? Then if we really wanted to check the `public` field we would call a `GetPublicOrPrivateImage` function instead?

Reply:
I'm not sure, feels OK in principle, although that would likely lead to just more `if` statements in the two places I can see where we accept a bool from somewhere (either user request or existing DB row) and then pass that into this function.